### ISSUE TITLE
allow_empty=False not effective on child ListSerializer during partial updates

### ISSUE THREAD

**Issue by reupen** [encode/django-rest-framework#6509]
## Checklist

- [x] I have verified that that issue exists against the `master` branch of Django REST framework.
- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] This is not a usage question. (Those should be directed to the [discussion group](https://groups.google.com/forum/#!forum/django-rest-framework) instead.)
- [x] This cannot be dealt with as a third party library. (We prefer new functionality to be [in the form of third party libraries](https://www.django-rest-framework.org/community/third-party-packages/#about-third-party-packages) where possible.)
- [x] I have reduced the issue to the simplest possible case.
- [ ] I have included a failing test as a pull request. (If you are unable to do so we can still accept the issue.)

## Steps to reproduce

Run this:

```python
from rest_framework import serializers

class ChildSerializer(serializers.Serializer):
    id = serializers.IntegerField()

class ParentSerializer(serializers.Serializer):
     ids = ChildSerializer(many=True, allow_empty=False)

serializer = ParentSerializer(data={'ids': []}, partial=True)
serializer.is_valid()
```

## Expected behavior

`serializer.is_valid()` should return False (as it does when `partial=False`) because empty values are not allowed.

## Actual behavior

It returns True.

I tracked it down to [this line](https://github.com/encode/django-rest-framework/blob/d2d1888217e8cc5aba995edf522fec903b1f91be/rest_framework/serializers.py#L652) and [this PR](https://github.com/encode/django-rest-framework/pull/4222).

I believe it an empty list should not be valid because `partial=True` is usually about being able to specify a subset of keys. If a value is provided it should still go through the usual validation (in the same way that `partial=True` does not override `allow_null=False`).

The documentation for `allow_empty` does not mention this behaviour either: https://www.django-rest-framework.org/api-guide/serializers/#allow_empty

**Comment by turfaa** [comment]
I agree with you. I would expect it to be not valid too. Should I open a PR? Will be a quick fix though.

**Comment by rpkilby** [comment]
I believe this was fixed in #6512.
